### ISSUE TITLE
Custom Attributes marked with RequiresPreviewFeatures

### ISSUE THREAD

**Issue by pgovind** [dotnet/roslyn-analyzers#5346]
Current CA2252: APIs marked with RequiresPreviewFeaturesAttribute doesn't handle custom attributes. For ex (taken from the TestCustomAttribute unit test):

``` csharp
using System.Runtime.Versioning; using System;
namespace Preview_Feature_Scratch
{

    class Program
    {
        static void Main(string[] args)
        {
            A anObject = new A();
        }
    }

[My]
class A
{
}

[RequiresPreviewFeatures]
[AttributeUsage(AttributeTargets.All)]
class MyAttribute : Attribute
{
}
}

```

I think the intent here is that usage of `[My]` should be marked with a diagnostic, but nothing else i.e. `A` is not a preview type, only `[My]` is. I see a couple of issues here:
1. Is there an easy way to mark `[|[My]|]` above the class definition of `A`? I'm not sure. cc @sharwell @jmarolf @mavasani 
2. Assuming 1 can be implemented, the only action a user can take is to suppress the warning around `[|[My]|]`. (In contrast, in other cases, an API/Type can be marked as Preview itself. In this instance, marking `A` as Preview wouldn't get rid of the diagnostic and worse, will make `A` a Preview type.)

**Comment by jeffhandley** [comment]
I _think_ we'd need to analyze everything that can have attributes on it, use `GetAttributes`, and then check the attribute types. That would be incredibly expensive with low value.  It could be optimized by keeping a list of preview attribute types, checking what their targets are, and only evaluating targets that could possibly be in scope.  But I think we can wait for a preview attribute scenario to emerge.

Attributes can have diagnostics on them using `ISymbol.GetAttributes()` and then for the `AttributeData` that is a violation, use `context.ReportDiagnostic(initializerAttribute.ApplicationSyntaxReference.GetSyntax().CreateDiagnostic(Rule));`.

**Comment by pgovind** [comment]
Will be fixed with https://github.com/dotnet/roslyn-analyzers/pull/5340
